File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    import math
    
    answer = []
    
    days = []
    for p, s in zip(progresses, speeds):
        days.append(math.ceil((100-p)/s))
        
    most = days[0]
    cnt = 0
    for day in days:
        if day > most:
            most = day
            answer.append(cnt)
            cnt = 1
        else:
            cnt += 1
    answer.append(cnt)
    
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    progresses = progresses.map((v, i) => Math.ceil((100 - v)/speeds[i]));
    
    while (progresses.length > 0) {
        let cnt = 1;
        let x = progresses.shift();
        while (progresses[0] <= x) {
            progresses.shift();
            cnt++;
        }
        answer.push(cnt);
    }
    return answer;
}